d2graph/seqdiagram.go

```diff
@@ -97,3 +97,10 @@ func (e *Edge) ContainedBy(obj *Object) bool {
 	}
 	return false
 }
+
+func (e *Edge) GetGroup() *Object {
+	for _, ref := range e.References {
+		return ref.ScopeObj
+	}
```

Comment:
that's not right. u might change the edge stroke color globally for example, in which case a reference's scope obj would be root.

u have to further check that the scopeobj is in sequence diagram.